pkg/controller/clusterinstallation/controller.go

```diff
@@ -0,0 +1,264 @@
+	}
+
+	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &mattermostv1alpha1.ClusterInstallation{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileClusterInstallation{}
+
+// ReconcileClusterInstallation reconciles a ClusterInstallation object
+type ReconcileClusterInstallation struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a ClusterInstallation object and makes changes based on the state read
+// and what is in the ClusterInstallation.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileClusterInstallation) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ClusterInstallation")
+
+	// Fetch the ClusterInstallation instance
+	mattermost := &mattermostv1alpha1.ClusterInstallation{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, mattermost)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	err = mattermost.SetDefaults()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if mattermost.Spec.DatabaseType.Type == "mysql" {
+		reqLogger.Info("Reconciling ClusterInstallation MySQL service account")
+		err = r.checkMySQLServiceAccount(mattermost, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		reqLogger.Info("Reconciling ClusterInstallation MySQL role binding")
+		err = r.checkMySQLRoleBinding(mattermost, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		reqLogger.Info("Reconciling ClusterInstallation MySQL")
+		err = r.checkMySQLDeployment(mattermost, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		reqLogger.Info("Reconciling ClusterInstallation Postgres")
+		err = r.checkDBPostgresDeployment(mattermost, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	// TODO
+	err = r.checkMinioDeployment(mattermost, reqLogger)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = r.checkMattermost(mattermost, reqLogger)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Already exists - don't requeue
```

Comment:
This comment confuses me a bit: it reads like an `else` case after a bunch of other conditions failed, but we reach it in all happy paths after potentially having done work. Can you help me understand the intent here?

Reply:
Not a comment I added, @cpanato any insight?

Reply:
was a comment added when we bootstrap. we can remove that